Tests/GRPCTests/GRPCReflectionServiceTests/GRPCReflectionServiceTests.swift

```diff
@@ -0,0 +1,241 @@
+
+    var iterator = serviceReflectionInfo.responseStream.makeAsyncIterator()
+    let message = try await iterator.next()
+    let receivedFileDescriptorProto =
+      try Google_Protobuf_FileDescriptorProto(serializedData: (
+        message?.fileDescriptorResponse
+          .fileDescriptorProto[0]
+      )!)
+    XCTAssertEqual(receivedFileDescriptorProto.name, self.fileDescriptorProto!.name)
+    XCTAssertEqual(
+      receivedFileDescriptorProto.service.count,
+      self.fileDescriptorProto!.service.count
+    )
+    XCTAssertEqual(
+      receivedFileDescriptorProto.service.first!.method.first!.name,
+      self.fileDescriptorProto!.service.first!.method.first!.name
+    )
+    XCTAssertEqual(message?.fileDescriptorResponse.fileDescriptorProto.count, 5)
+  }
+
+  func testListServices() async throws {
+    try self.setUpServerAndChannel()
+    let client = Reflection_ServerReflectionAsyncClient(channel: self.channel!)
+    let serviceReflectionInfo = client.makeServerReflectionInfoCall()
+    try await serviceReflectionInfo.requestStream.send(.with {
+      $0.host = "127.0.0.1"
+      $0.listServices = "services"
+    })
+    serviceReflectionInfo.requestStream.finish()
+    var iterator = serviceReflectionInfo.responseStream.makeAsyncIterator()
+    let message = try await iterator.next()
+    let receivedServices = message?.listServicesResponse.service.map { $0.name }
+    XCTAssertEqual(receivedServices?.count, 5)
+    for serviceName in receivedServices! {
+      XCTAssertTrue(self.servicesNames.contains(serviceName))
+    }
+  }
+
+  func testReflectionServiceData() throws {
+    self.createFileDescriptorProto()
+    do {
+      let registry = try ReflectionServiceData(fileDescriptorProtos: self.protos)
+      let registryFileDescriptorData = registry.getfileDescriptorData()
+      let services = registry.getServices()
+
+      for (fileName, protoData) in registryFileDescriptorData {
+        let serializedFiledescriptorData = protoData.getSerializedFileDescriptorProto()
+        let dependecies = protoData.getDependency()
+
+        let originalIndex = self.protos.firstIndex(where: { $0.name == fileName })
+        XCTAssertNotNil(originalIndex)
+
+        let originalProto = self.protos[originalIndex!]
+        XCTAssertEqual(originalProto.name, fileName)
+        XCTAssertEqual(try originalProto.serializedData(), serializedFiledescriptorData)
+        XCTAssertEqual(originalProto.dependency, dependecies)
+
+        self.protos.remove(at: originalIndex!)
+      }
+
+      for serviceName in services {
+        XCTAssert(self.servicesNames.contains(serviceName))
+        self.servicesNames.removeAll { $0 == serviceName }
+      }
+    } catch {
+      XCTFail(error.localizedDescription)
+    }
+  }
+
+  func testGetSerializedFileDescriptorProtos() throws {
+    self.createFileDescriptorProto()
+    do {
+      let registry = try ReflectionServiceData(fileDescriptorProtos: self.protos)
```

Comment:
You need more test cases than this covering different dependency graphs. We also need a test for getting the file descriptor protos of a filename we don't know about.